module/extension/Director/src/Director.cpp

```diff
@@ -213,24 +214,19 @@ namespace module::extension {
             pack_builder.erase(done.requester_task_id);
         });
 
-        // A state that we were monitoring is updated, we might be able to run the reaction now
+        // A state that we were monitoring is updated, we might be able to run the task now
         on<Trigger<StateUpdate>, Sync<Director>>().then("State Updated", [this](const StateUpdate& update) {
-            // TODO ALGORITHM SECTION HERE
-            // This state update event means the conditions for the Provider specified in the update are now met
-        });
-
-        // We do things when we receive tasks to run
-        on<Trigger<TaskPack>, Sync<Director>>().then("Run Task Pack", [this](const TaskPack& pack) {
-            // TODO ALGORITHM SECTION HERE
-            // We have just been given a new task pack
+            // Get the group that had a state update
+            auto p  = providers[update.provider_id];
+            auto& g = groups[p->type];
 
-            // TODO check if there are tasks that were previously in the pack from this Provider and now are gone
-            // For those we need to end execution of those tasks (and their children)
+            // Go check if this state update has changed any of the tasks that are queued
+            reevaluate_queue(g);
+        });
 
-            for (const auto& task : pack) {
-                // TODO run this task through the director's algorithm to work out what to do with it
-                // TODO we need to process this pack as a whole, if one of the required actions fail they all do
-            }
+        // We have a new task pack to run
+        on<Trigger<TaskList>, Sync<Director>>().then("Run Task Pack", [this](const TaskList& pack) {  //
+            run_task_pack(pack);
```

Comment:
```suggestion
            run_task_list(list);
```

Reply:
likewise, I want to separate the concept of a task list and a task pack